CMakeLists.txt

```diff
@@ -11,7 +11,7 @@ set(ANDROID_MKE2FS_NAME "mke2fs.android")
 
 # Version of android-tools and the version of boringssl being used.
 # See: https://android.googlesource.com/platform/external/boringssl/+/platform-tools-${ANDROID_VERSION}/BORINGSSL_REVISION
-set(ANDROID_VERSION 30.0.5p1)
+set(ANDROID_VERSION 31.0.0)
 set(BORINGSSL_VERSION 2fb729d4f36beaf263ad85e24a790b571652679c)
```

Comment:
https://android.googlesource.com/platform/external/boringssl/+/platform-tools-31.0.0/BORINGSSL_REVISION

Seems to me we need to update this to `c1e156ae16fa8b61af9b5d2b74e59d3f86e141be`.

Reply:
Oh yeah, totally forgot about boringssl

Reply:
Is there any way to check at build-time (in CMake files) that the boringssl tree corresponds to file BORINGSSL_REVISION?

`rev=$(cd vendor/boringssl; git rev-list -n 1 HEAD)`

Such automatic check would eliminate such issues in the future.

Reply:
Probably is with [execute_process](https://cmake.org/cmake/help/latest/command/execute_process.html). Wouldn't do that as part of this PR though.